src/common/src/array/jsonb_array.rs

```diff
@@ -69,29 +70,43 @@ impl ArrayBuilder for JsonbArrayBuilder {
         for bit in other.bitmap.iter() {
             self.bitmap.append(bit);
         }
-        self.data.extend_from_slice(&other.data);
+        for value in other.data.as_array().unwrap().iter() {
+            self.builder.add_value(value);
+        }
     }
 
     fn pop(&mut self) -> Option<()> {
-        self.data.pop().map(|_| self.bitmap.pop().unwrap())
+        self.bitmap.pop();
+        todo!("pop from jsonb builder");
```

Comment:
Is the function be called?

Reply:
I found the only place where the pop function is called was introduced by your PR #5260. 🤡
Looks like it is reachable. I'll fix this later.